sdcm/cluster_baremetal.py

```diff
@@ -10,16 +12,34 @@
 MONITOR_NAME = 'monitor-node'
 
 
+class NodeInfo(TypedDict):
+    public_ip: str
+    private_ip: str
+
+
+class NodeCredentialInformation(TypedDict):
+    username: str
+    node_list: list[NodeInfo]
+
+
+class BareMetalCredentials(TypedDict):
+    db_nodes: NodeCredentialInformation
+    loader_nodes: NodeCredentialInformation
+    monitor_nodes: NodeCredentialInformation
+
+
 class NodeIpsNotConfiguredError(Exception):
     pass
 
 
 class PhysicalMachineNode(cluster.BaseNode):
     log = LOGGER
 
-    def __init__(self, name, parent_cluster, public_ip, private_ip, credentials, base_logdir=None, node_prefix=None):  # pylint: disable=too-many-arguments
+    # pylint: disable=too-many-arguments
+    def __init__(self, name, parent_cluster: 'PhysicalMachineCluster',
+                 public_ip, private_ip, credentials, base_logdir=None, node_prefix=None):
         ssh_login_info = {'hostname': None,
-                          'user': credentials.name,
+                          'user': getattr(self.parent_cluster, "ssh_username", credentials.name),
```

Comment:
We have a problem here: `self.parent_cluster` will be assigned later in `super().__init__()`, need to change it to just `parent_cluster`. I think it's came from my example, sorry about that :(